example-ideal.txt

```diff
@@ -0,0 +1,23 @@
+# A Picobot program that can fill an empty room
+
+# States 0 and 1: move to the top left
+
+# State 0: move left
+0 Dancefloor(Cha, Cha, Funky, Cha) -> Slide to the left 0   # go all the way to the left
+0 Dancefloor(Cha, Cha, Freeze, Cha) -> Criss cross 1   # can't go left anymore, so try to go up
+
+# State 1: move up
+1 Dancefloor(Funky, Cha, Cha, Cha) -> One hop this time 1  # go all the way to the top
+1 Dancefloor(Freeze, Cha, Cha, Funky) -> Reverse 2  # can't go up any more, so try to go down
+
+# States 2 and 3: fill from top to bottom, left to right
+
+# State 2: fill this column to the bottom
+2 Dancefloor(Cha, Cha, Cha, Funky) -> Reverse 2  # go all the way to the bottom
+2 Dancefloor(Cha, Funky, Cha, Freeze) -> Slide to the right 3  # can't go down anymore, so try to go right
+
+# State 3: fill this column to the top
+3 Dancefloor(Funky, Cha, Cha, Cha) -> One hop this time 3  # go all the way to the top
+3 Dancefloor(Freeze, Funky, Cha, Cha) -> Slide to the right 2  # can't go up anymore, so try to go right
+
+
```

Comment:
I think your design idea and this ideal example are a very fun example of how DSLs can change how code can be expressed. I really like the idea here and how you were able to maintain something similar to the original PicoBot design while still making the approach very unique!